src/devices/Common/Iot/Device/Graphics/BdfFont.cs

```diff
@@ -0,0 +1,266 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Graphics
+{
+    public class BdfFont
+    {
+        public int Width  { private set; get; }
+        public int Height { private set; get; }
+        public int XDisplacement { private set; get; }
+        public int YDisplacement { private set; get; }
+        public int DefaultChar  { private set; get; }
+        public int CharsCount   { private set; get; }
+
+        private Dictionary<int, int> GlyphMapper { set; get; }
+        private int BytesPerGlyph { set; get; }
+        private ushort[] GlyphUshortData { set; get; }
+
+        private static readonly string s_fontBoundingBox    = "FONTBOUNDINGBOX ";
+        private static readonly string s_charSet            = "CHARSET_REGISTRY ";
+        private static readonly string s_isoCharset         = "\"ISO10646\"";
+        private static readonly string s_defaultChar        = "DEFAULT_CHAR ";
+        private static readonly string s_Chars              = "CHARS ";
+        private static readonly string s_startChar          = "STARTCHAR ";
+        private static readonly string s_encoding           = "ENCODING ";
+        // private static readonly string s_sWidth             = "SWIDTH";
+        // private static readonly string s_dWidth             = "DWIDTH";
+        private static readonly string s_bbx                = "BBX ";
+        // private static readonly string s_vVector            = "VVECTOR";
+        private static readonly string s_endChar            = "ENDCHAR";
+        private static readonly string s_bitmap             = "BITMAP";
+
+        private BdfFont() { }
+
+        public static BdfFont Load(string fontFilePath)
+        {
+            using (StreamReader sr = new StreamReader(fontFilePath))
+            {
+                BdfFont font = new BdfFont();
+                while (!sr.EndOfStream)
+                {
+                    ReadOnlySpan<char> span = sr.ReadLine().AsSpan().Trim();
+                    if (span.StartsWith(s_fontBoundingBox.AsSpan(), StringComparison.Ordinal))
+                    {
+                        span = span.Slice(s_fontBoundingBox.Length).Trim();
+                        font.Width = ReadNextDecimalNumber(ref span);
+                        font.Height = ReadNextDecimalNumber(ref span);
+                        font.XDisplacement = ReadNextDecimalNumber(ref span);
+                        font.YDisplacement = ReadNextDecimalNumber(ref span);
+                        font.BytesPerGlyph = (int) Math.Ceiling(((double)font.Width) / 8);
+                    }
+                    else if (span.StartsWith(s_charSet.AsSpan(), StringComparison.Ordinal))
```

Comment:
is `AsSpan` needed here? feels like this should get implicitly converted

Reply:
Yes we can get rid of AsSpan.